pyomeca/markers.py

```diff
@@ -1,180 +1,513 @@
-        Parameters
-        ----------
-        m : np.array
-            A CSV tabular matrix (Fx3*N)
-        Returns
-        -------
-        Vectors3d of data set
+        Returns:
+            Random markers `xarray.DataArray` sampled from a given distribution
+
+        !!! example
+            To instantiate a `Markers` with some random data sampled from a normal distribution:
+
+            ```python
+            from pyomeca import Markers
+
+            n_axis = 3
+            n_channels = 10
+            n_frames = 100
+            size = n_axis, n_channels, n_frames
+            markers = Markers.from_random_data(size=size)
+            ```
+
+            You can choose any distribution available in
+                [numpy.random](https://docs.scipy.org/doc/numpy-1.14.0/reference/routines.random.html#distributions):
+
+            ```python
+            markers = Markers.from_random_data(distribution="uniform", size=size, low=1, high=10)
+            ```
         """
-        s = m.shape
-        if s[1] % 3 != 0:
-            raise IndexError("Number of columns must be divisible by 3")
-        return Markers3d(np.reshape(m.T, (3, int(s[1] / 3), s[0]), "F"))
+        return Markers(getattr(np.random, distribution)(size=size, **kwargs).cumsum(-1))
 
     @classmethod
-    def from_trc(cls, filename):
-        return cls.from_csv(
+    def from_csv(
+        cls,
+        filename: Union[str, Path],
+        usecols: Optional[List[Union[str, int]]] = None,
+        header: Optional[int] = None,
+        first_row: int = 0,
+        first_column: Optional[Union[str, int]] = None,
+        time_column: Optional[Union[str, int]] = None,
+        last_column_to_remove: Optional[Union[str, int]] = None,
+        prefix_delimiter: Optional[str] = None,
+        suffix_delimiter: Optional[str] = None,
+        skiprows: Optional[List[int]] = None,
+        pandas_kwargs: Optional[dict] = None,
+        attrs: Optional[dict] = None,
+    ) -> xr.DataArray:
+        """
+        Markers DataArray from a csv file.
+
+        Arguments:
+            filename: Any valid string path
+            usecols: All elements must either be positional or strings that correspond to column names.
+                For example, a valid list-like usecols parameter would be [0, 1, 2] or ['foo', 'bar', 'baz'].
+            header: Row of the header (0-indexed)
+            first_row: First row of the data (0-indexed)
+            first_column: First column of the data (0-indexed)
+            time_column: Location of the time column. If None, indices are associated
+            last_column_to_remove: If for some reason the csv reads extra columns, how many should be ignored
+            prefix_delimiter: Delimiter that split each column name by its prefix (we keep only the column name)
+            suffix_delimiter: Delimiter that split each column name by its suffix (we keep only the column name)
+            skiprows: Line numbers to skip (0-indexed)
```

Comment:
If we keep skiprows, should this be `skip_rows`?

Reply:
Yes, let's be consistent :)
Done in https://github.com/pyomeca/pyomeca/pull/88/commits/b6dc17e1b619fed6280e91b845e4433e9617dec3.